File: solutions/python/bob/1/bob.py
import re

def response(txt):
    if txt.strip()=='':
         return 'Fine. Be that way!'
    
    if upperconnword(txt) and txt.strip()[-1]=='?':
         return 'Calm down, I know what I\'m doing!'
    
    if txt.strip()[-1]=='?':
         return 'Sure.'
    
    if upperconnword(txt):
         return 'Whoa, chill out!'

    return 'Whatever.'

    


def upperconnword(txt):
     return txt.strip().isupper()

File: solutions/javascript/bob/1/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  const trimmedMessage = message.trim();

  if (trimmedMessage === '') {
    return 'Fine. Be that way!';
  }

  const isYelling = trimmedMessage === trimmedMessage.toUpperCase() && /[A-Z]/.test(trimmedMessage);
  const isQuestion = trimmedMessage.endsWith('?');

  if (isYelling && isQuestion) {
    return 'Calm down, I know what I\'m doing!';
  } else if (isYelling) {
    return 'Whoa, chill out!';
  } else if (isQuestion) {
    return 'Sure.';
  } else {
    return 'Whatever.';
  }
};
